src/NodaTime/XmlSchemaDefinition.cs

```diff
@@ -0,0 +1,241 @@
+        internal static XmlQualifiedName AddLocalDateSchemaType(XmlSchemaSet schemaSet) => AddSchemaType(schemaSet, LocalDateSchemaType.Value);
+        internal static XmlQualifiedName AddLocalDateTimeSchemaType(XmlSchemaSet schemaSet) => AddSchemaType(schemaSet, LocalDateTimeSchemaType.Value);
+        internal static XmlQualifiedName AddLocalTimeSchemaType(XmlSchemaSet schemaSet) => AddSchemaType(schemaSet, LocalTimeSchemaType.Value);
+        internal static XmlQualifiedName AddOffsetSchemaType(XmlSchemaSet schemaSet) => AddSchemaType(schemaSet, OffsetSchemaType.Value);
+        internal static XmlQualifiedName AddOffsetDateSchemaType(XmlSchemaSet schemaSet) => AddSchemaType(schemaSet, OffsetDateSchemaType.Value);
+        internal static XmlQualifiedName AddOffsetDateTimeSchemaType(XmlSchemaSet schemaSet) => AddSchemaType(schemaSet, OffsetDateTimeSchemaType.Value);
+        internal static XmlQualifiedName AddOffsetTimeSchemaType(XmlSchemaSet schemaSet) => AddSchemaType(schemaSet, OffsetTimeSchemaType.Value);
+        internal static XmlQualifiedName AddPeriodBuilderSchemaType(XmlSchemaSet schemaSet) => AddSchemaType(schemaSet, PeriodBuilderSchemaType.Value);
+        internal static XmlQualifiedName AddYearMonthSchemaType(XmlSchemaSet schemaSet) => AddSchemaType(schemaSet, YearMonthSchemaType.Value);
+        internal static XmlQualifiedName AddZonedDateTimeSchemaType(XmlSchemaSet schemaSet) => AddSchemaType(schemaSet, ZonedDateTimeSchemaType.Value);
+
+        private const string YearPattern = @"-?[0-9]{4}";
+        private const string MonthPattern = @"[0-9]{2}";
+        private const string DayPattern = @"[0-9]{2}";
+        private const string TimePattern = @"[0-9]{2}:[0-9]{2}:[0-9]{2}(\.[0-9]{1,9})?";
+        private const string OffsetPattern = @"(Z|[+-][0-9]{2}(:[0-9]{2}(:[0-9]{2})?)?)";
+        private const string PeriodBuilderPattern = @"P(-?[0-9]+Y)?(-?[0-9]+M)?(-?[0-9]+W)?(-?[0-9]+D)?(T(-?[0-9]+H)?(-?[0-9]+M)?(-?[0-9]+S)?(-?[0-9]+s)?(-?[0-9]+t)?(-?[0-9]+n)?)?";
+        private const string DurationPattern = @"-?[0-9]{1,8}:[0-9]{2}:[0-9]{2}:[0-9]{2}(\.[0-9]{1,9})?";
+
+        static XmlSchemaDefinition()
+        {
+            var xsStringType = XmlSchemaType.GetBuiltInSimpleType(XmlTypeCode.String);
+
+            var calendarRestriction = new Lazy<XmlSchemaType>(() => CreateEnumerationRestriction("calendar", xsStringType, CalendarSystem.Ids));
+            var localDateRestriction = new Lazy<XmlSchemaType>(() => CreatePatternRestriction("localDate", xsStringType, $"{YearPattern}-{MonthPattern}-{DayPattern}"));
+            var localDateTimeRestriction = new Lazy<XmlSchemaType>(() => CreatePatternRestriction("localDateTime", xsStringType,  $"{YearPattern}-{MonthPattern}-{DayPattern}T{TimePattern}"));
+            var zoneRestriction = new Lazy<XmlSchemaType>(() => CreateEnumerationRestriction("zone", xsStringType, DateTimeZoneProviders.Serialization.GetAllZones().Select(e => e.Id)));
```

Comment:
As before, this makes me really nervous.

Reply:
Do you mean that like time zone ids, the calendar ids may change over time?

Reply:
Nope, this was a reiteration of the zone ID concern.